vrpn_Oculus.C

```diff
@@ -0,0 +1,337 @@
+/** @file	vrpn_Oculus.C
+@brief	Drivers for various Oculus devices.  Initially, just the DK2.
+
+  @date 2015
+  @copyright 2015 ReliaSolve.com
+  @author ReliaSolve.com russ@reliasolve.com
+  @license Standard VRPN license.
+*/
+
+// Based on the OSVR hacker dev kit driver by Kevin Godby.
+// Based on Oliver Kreylos' OculusRiftHIDReports.cpp and OculusRift.cpp.
+
+#include "vrpn_Oculus.h"
+
+#include "vrpn_BaseClass.h"  // for ::vrpn_TEXT_NORMAL, etc
+
+VRPN_SUPPRESS_EMPTY_OBJECT_WARNING()
+
+#if defined(VRPN_USE_HID)
+
+// USB vendor and product IDs for the models we support
+static const vrpn_uint16 OCULUS_VENDOR = 0x2833;
+static const vrpn_uint16 DK2_PRODUCT = 0x0021;
+
+vrpn_Oculus_DK2::vrpn_Oculus_DK2(bool enableLEDs,
+  const char *name, vrpn_Connection *c,
+  double keepAliveSeconds)
+    : vrpn_Analog(name, c)
+    , vrpn_HidInterface(m_filter =
+      new vrpn_HidProductAcceptor(OCULUS_VENDOR, DK2_PRODUCT))
+{
+  d_keepAliveSeconds = keepAliveSeconds;
+  d_enableLEDs = enableLEDs;
+
+  if (enableLEDs) {
+    vrpn_Analog::num_channel = 12;
+  } else {
+    vrpn_Analog::num_channel = 11;
+  }
+  memset(channel, 0, sizeof(channel));
+  memset(last, 0, sizeof(last));
+
+  // Set the timestamp
+  vrpn_gettimeofday(&d_timestamp, NULL);
+  d_lastKeepAlive = d_timestamp;
+}
+
+vrpn_Oculus_DK2::~vrpn_Oculus_DK2()
+{
+  delete m_acceptor;
+}
+
+// Copied from Oliver Kreylos' OculusRift.cpp.
```

Comment:
Asked him in email if he'd be willing to release this function and the know-how under BSL1.0.  We should wait until we hear back before pulling, perhaps, otherwise we either need to do a clean rewrite or else release this file under GPL.  We could go GPL now and fix later, I suppose.